File: archive/c/c-sharp/RemoveAllWhitespace.cs
if (args is not [var input] || string.IsNullOrEmpty(input))
    return ExitWithError();

RemoveWhitespace(input.AsSpan());
return 0;

static void RemoveWhitespace(ReadOnlySpan<char> input)
{
    char[] buffer = new char[input.Length];
    int j = 0;

    foreach (char c in input)
    {
        if (!char.IsWhiteSpace(c))
            buffer[j++] = c;
    }

    Console.WriteLine(new string(buffer, 0, j));
}

static int ExitWithError()
{
    Console.WriteLine("Usage: please provide a string");
    return 1;
}

File: archive/v/visual-basic/remove-all-whitespace.vb
Public Module RemoveAllWhitespace

    Private Sub ShowUsage()
        Console.WriteLine("Usage: please provide a string")
        Environment.Exit(1)
    End Sub


    Private Sub RemoveAllWhitespace(input As String)

        Dim result As New System.Text.StringBuilder()

        For Each ch As Char In input

            If Not Char.IsWhiteSpace(ch) Then
                result.Append(ch)
            End If

        Next

        Console.WriteLine(result.ToString())

    End Sub


    Public Function Main(args As String()) As Integer

        If args Is Nothing OrElse args.Length = 0 OrElse String.IsNullOrEmpty(args(0)) Then
            ShowUsage()
        End If

        RemoveAllWhitespace(args(0))

        Return 0

    End Function

End Module